internal/core/entities/scheduler.go

```diff
@@ -202,3 +204,25 @@ func (s *Scheduler) HasValidPortRangeConfiguration() error {
 
 	return nil
 }
+
+func (s *Scheduler) HasMajorVersionDifference(otherVersion string) bool {
+	return extractMajorVersion(otherVersion) != extractMajorVersion(s.Spec.Version)
```

Comment:
You can use the `semver` lib and just check the major difference, like [here](https://github.com/topfreegames/maestro/blob/4fe0422078ef7a1144dca887397fae06a8561250/internal/core/operations/schedulers/newversion/executor.go#L310)

Reply:
fixed to use this lib